### PR TITLE
Added support for async uploads

### PR DESCRIPTION
**Pull request by gavinwilliams**
Async upload support for big files. As per code from #482

### CODE REVIEW

File: src/TwitterOAuth.php [hunk 1]
```diff
@@ -252,6 +252,20 @@ private function uploadMediaNotChunked($path, array $parameters)
...
 
+/**
+     * Checks to see whether the upload  was successful or not
+     *
+     * @param $media_id
+     * @return array|object
+     */
+    public function getUploadStatus($media_id)
+    {
+        return $this->http('GET', self::UPLOAD_HOST, 'media/upload', array(
```
Comment (abraham) [inline src/TwitterOAuth.php]: Use `[]` instead of `array()`

File: src/TwitterOAuth.php [hunk 2]
```diff
@@ -252,6 +252,20 @@ private function uploadMediaNotChunked($path, array $parameters)
         return $this->http('POST', self::UPLOAD_HOST, $path, $parameters);
     }
 
+/**
+     * Checks to see whether the upload  was successful or not
+     *
+     * @param $media_id
+     * @return array|object
+     */
+    public function getUploadStatus($media_id)
```
Comment (abraham) [inline src/TwitterOAuth.php]: Rename method to `uploadMediaStatus`

File: src/TwitterOAuth.php [hunk 3]
```diff
@@ -260,14 +274,20 @@ private function uploadMediaNotChunked($path, array $parameters)
      *
      * @return array|object
      */
-    private function uploadMediaChunked($path, array $parameters)
+    private function uploadMediaChunked($path, $parameters)
```
Comment (abraham) [inline src/TwitterOAuth.php]: Discard `uploadMediaChunked` changes in favor of #542

**Comment by cerber717** [comment]
👍  waiting for this one :)

**Comment by abraham** [comment]
I think this has all been covered with other PRs.

**Pull request closed without merging**